docs/reference/func_build.md

```diff
@@ -0,0 +1,68 @@
+## func build
+
+Build a function project as a container image
+
+### Synopsis
+
+Build a function project as a container image
+
+This command builds the function project in the current directory or in the directory
+specified by --path. The result will be a container image that is pushed to a registry.
+The func.yaml file is read to determine the image name and registry.
+If the project has not already been built, either --registry or --image must be provided
+and the image name is stored in the configuration file.
+
+
+```
+func build [flags]
+```
+
+### Examples
+
+```
+
+# Build from the local directory, using the given registry as target.
+# The full image name will be determined automatically based on the
+# project directory name
+func build --registry quay.io/myuser
+
+# Build from the local directory, specifying the full image name
+func build --image quay.io/myuser/myfunc
+
+# Re-build, picking up a previously supplied image name from a local func.yml
+func build
+
+# Build using s2i instead of Buildpacks
+func build --builder=s2i
+
+# Build with a custom buildpack builder
+func build --builder=pack --builder-image cnbs/sample-builder:bionic
+
+```
+
+### Options
+
+```
+  -b, --builder string         build strategy to use when creating the underlying image. Currently supported build strategies are "pack" or "s2i". (default "pack")
+      --builder-image string   builder image, either an as a an image name or a mapping name.
+                               Specified value is stored in func.yaml (as 'builder' field) for subsequent builds. ($FUNC_BUILDER_IMAGE)
+  -c, --confirm                Prompt to confirm all configuration options (Env: $FUNC_CONFIRM)
+  -h, --help                   help for build
+  -i, --image string           Full image name in the form [registry]/[namespace]/[name]:[tag] (optional). This option takes precedence over --registry (Env: $FUNC_IMAGE)
+  -p, --path string            Path to the project directory (Env: $FUNC_PATH) (default "/Users/lball/src/github.com/knative-sandbox/kn-plugin-func")
```

Comment:
@lance the default --path seems to be odd.
This happens in multiple files.

Reply:
This is probably not related to this PR.

Reply:
that's in cmd/root.go

Reply:
maybe we should replace `cwd()` with `""` and mention that current dir is used if --path is empty.

Reply:
`cmd.Flags().StringP("path", "p", "", "Path to the project directory if empty current dir is used (Env: $FUNC_PATH)")`

Reply:
I didn't even notice that before - good catch. That's happening here: https://github.com/knative-sandbox/kn-plugin-func/blob/3ac5d468bfdbb326375ebc5bc640729b59fec7f2/cmd/root.go#L336-L339

I wonder if we can determine runtime context and instead of `cwd()` when generating documentation, use something like `"$CWD"` or `"<current working directory>"`

Reply:
> maybe we should replace `cwd()` with `""` and mention that current dir is used if --path is empty.

I prefer this method if possible as well